### PR TITLE
fix: handling of backend options if using with other backends

### LINKED ISSUE #854: Default namespace not found at "/public/static/locales/en/common.json" when using Locize backend
## Describe the bug
You must provide a clear and concise description of what the bug is.

When backend (Locize) is provided, it is expected to find the namespace from the provided backend. However, its trying to search inside the /public/static folder.

## Occurs in next-i18next version
You must provide the npm module version, git tag, or even better the explict commit causing the issue.

"next-i18next": 6.0.3

## Steps to reproduce
You must provide a reproducible repository, or copy/paste blocks of code which are sufficient to reproduce the bug.

1. npx create-next-app
2. Add i18n.js with LocizeBackend
```
const NextI18Next = require('next-i18next').default
const LocizeBackend = require('i18next-locize-backend');

module.exports = new NextI18Next({
  use: [
    LocizeBackend
  ],
  backend: {
    projectId: PROJECT_ID
    apiKey: API_KEY
    version: process.env.LOCIZE_VERSION || 'latest',
  },
})
```
3. Update _app.js
```
import '../styles/globals.css'

import { appWithTranslation} from '../i18n'

function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}

export default appWithTranslation(MyApp)
```

## Expected behaviour
You must provide a clear and concise description of what you expected to happen.

It should import the namespace and translations from the backend

## Screenshots
If applicable, add screenshots or a GIF to help explain your problem.
image

## OS (please complete the following information)
 - Device: [e.g. MBP 2018 13"]
 - Browser: [e.g. Chrome 70.0.3538.77]

## Additional context

Tried to follow https://stackoverflow.com/questions/55994799/how-to-integrate-next-i18next-nextjs-locize, @adrai's https://github.com/isaachinman/next-i18next/pull/705, but not working.

### PR DESCRIPTION
**Pull request by adrai**
fixes #854

### CODE REVIEW

File: src/config/create-config.ts [hunk 1]
```diff
@@ -44,7 +42,8 @@ export const createConfig = (userConfig) => {
     return combinedConfig
   }
 
-  if (isServer()) {
+  const hasCustomBackend = combinedConfig.use && combinedConfig.use.find((b) => b.type === 'backend')
```
Comment (isaachinman) [inline src/config/create-config.ts:45]: Why check `combinedConfig` instead of `userConfig`?

Comment (adrai) [reply src/config/create-config.ts:45]: 🤷‍♂️ does it make any difference?
Do you want me to change it?

Comment (isaachinman) [reply src/config/create-config.ts:45]: If the `defaultConfig` ever contained `type === 'backend'`, this condition would be true 100% of the time. Also keep in mind that we don't deep merge the backend object until [the very end](https://github.com/isaachinman/next-i18next/blob/1d352541c543e0f08fc7bda724bf1f2c24ec0683/src/config/create-config.ts#L4).

**Comment by isaachinman** [comment]
Thanks @adrai. It occurs to me that the `combinedConfig.preload = allLanguages` line might still be important for custom backends. We rely on all locales/translations being available from init to be able to handle SSR. What do you think?

**Comment by adrai** [comment]
Do you want me to move that line after line 27?

**Comment by adrai** [comment]
@isaachinman What's the problem with the node-multi-build ? I can't see the details.

**Comment by adrai** [comment]
> @isaachinman What's the problem with the node-multi-build ? I can't see the details.

found it

**Comment by isaachinman** [comment]
@adrai It's actual test failures. If you run `yarn test` locally, you'll see it.

**Comment by adrai** [comment]
> @adrai It's actual test failures. If you run `yarn test` locally, you'll see it.

had to clean node_modules, idkw...
now it's working... push is coming in next minutes

**Comment by adrai** [comment]
@isaachinman ok for you? btw: have you seen the change regarding supportedLngs?

File: src/config/create-config.ts [hunk 2]
```diff
@@ -34,7 +38,12 @@ export const createConfig = (userConfig) => {
     localeStructure,
   } = combinedConfig
 
-
+  if (isServer()) {
```
Comment (isaachinman) [inline src/config/create-config.ts:41]: Sorry, last nitpick, can this go inside the existing `isServer` block, and have a nested `!hasCustomBackend`? Seems a bit more readable/maintainable to me.

Comment (adrai) [reply src/config/create-config.ts:41]: The poblem is, the else block should be executed too

Comment (adrai) [reply src/config/create-config.ts:41]: or not?

Comment (adrai) [reply src/config/create-config.ts:41]: maybe not

**Comment by adrai** [comment]
Now, codeclimate has an issue: https://codeclimate.com/github/isaachinman/next-i18next/pull/861 any idea?

File: src/config/create-config.ts (hunk 2 above)
Comment (isaachinman) [reply src/config/create-config.ts]: Sorry, I don't get what you mean about the `else` block? We should just wrap the code that we _don't_ want to run inside of the `isServer` block inside a secondary `if (!hasCustomBackend)` condition.

Comment (adrai) [reply src/config/create-config.ts]: already done this

**Comment by adrai** [comment]
@isaachinman did you have a chance to look at it?

**Comment by isaachinman** [comment]
@adrai CodeClimate approved. Hopefully I will have time for a final review by end of week. Thanks so much!

**Comment by dkleber89** [comment]
Would be nice when this bugfix get merged soon ... Thanks a lot guys

**Review by isaachinman — APPROVED** [review]

**Pull request merged**